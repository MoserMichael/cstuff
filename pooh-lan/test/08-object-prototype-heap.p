include 'heap.inc'

h = make_min_heap()

for i range( ~from 3 ~to 1 ~step -1 )
    h . push( ~key i ~data  i ) # '[i]' )
end

while ! h.empty()
    [key, val ] = h.pop()

    println( ~msg 'key [ key ] value [ val ]' )
end
