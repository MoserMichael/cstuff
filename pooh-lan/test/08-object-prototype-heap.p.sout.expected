key 1 value 1
key 2 value 2
key 3 value 3
